language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/signature/SignatureHelpUtil.java

```diff
@@ -92,21 +92,35 @@ private SignatureHelpUtil() {
         SignatureInfoModel signatureInfoModel = getSignatureInfoModel(functionSymbol.get(), context);
 
         // Override label for 'new' constructor
-        String label = functionSymbol.get().name();
-        int initIndex = label.indexOf(INIT_SYMBOL);
-        if (initIndex > -1) {
-            label = "new " + label.substring(0, initIndex);
-        }
+        int initIndex = functionSymbol.get().name().indexOf(INIT_SYMBOL);
+        StringBuilder labelBuilder = initIndex > -1
+                ? new StringBuilder("new " + functionSymbol.get().name().substring(0, initIndex))
+                : new StringBuilder(functionSymbol.get().name());
 
+        labelBuilder.append("(");
         // Join the function parameters to generate the function's signature
-        String paramsJoined = signatureInfoModel.getParameterInfoModels().stream().map(parameterInfoModel -> {
-            // For each of the parameters, create a parameter info instance
-            parameterInformationList.add(getParameterInformation(parameterInfoModel));
-
-            return parameterInfoModel.toString();
-        }).collect(Collectors.joining(", "));
+        List<ParameterInfoModel> parameterInfoModels = signatureInfoModel.getParameterInfoModels();
+        for (ParameterInfoModel paramModel : parameterInfoModels) {
+            int labelOffset = labelBuilder.toString().length();
+            labelBuilder.append(paramModel.parameter.getType());
+            ParameterInformation paramInfo = new ParameterInformation();
+            paramInfo.setDocumentation(getParameterDocumentation(paramModel));
+            int paramStart = labelOffset;
+            int paramEnd = labelOffset + paramModel.parameter.getType().length();
+            if (paramModel.parameter.getName().isPresent()) {
+                paramStart = paramEnd + 1;
+                paramEnd += (paramModel.parameter.getName().get() + " ").length();
+                labelBuilder.append(" ").append(paramModel.parameter.getName().get());
+            }
+            if (parameterInfoModels.indexOf(paramModel) < parameterInfoModels.size() - 1) {
+                labelBuilder.append(", ");
```

Comment:
we can get rid of `indexOf` call for each element if we move the parent loop into an indexed for loop. Or will be more readable if we can move parameters generation into a `new StringJoiner(", ")`.

Reply:
converted to an indexed loop